engine/src/main/java/org/terasology/engine/subsystem/common/TelemetrySubSystem.java

```diff
@@ -0,0 +1,125 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.config.Config;
+import org.terasology.context.Context;
+import org.terasology.engine.subsystem.EngineSubsystem;
+import org.terasology.registry.In;
+import org.terasology.registry.InjectionHelper;
+import org.terasology.telemetry.Metrics;
+import org.terasology.telemetry.TelemetryParams;
+import org.terasology.telemetry.TelemetryUtils;
+
+import java.util.List;
+
+/**
+ * This is a telemetry engine system.
+ */
+public class TelemetrySubSystem implements EngineSubsystem {
+
+    @In
+    private Config config;
+
+    private Metrics metrics;
+
+    private Emitter emitter;
+
+    // Name space to identify the tracker
+    private final String NAMESPACE_TRACKER = this.getClass().toString();
+
+    private static final Logger logger = LoggerFactory.getLogger(TelemetrySubSystem.class);
+
+    @Override
+    public String getName() {
+        return "Telemetry";
+    }
+
+    @Override
+    public void preInitialise(Context rootContext) {
+
+        // add metrics to context, this helps show metric values in ui
+        metrics = new Metrics();
+        rootContext.put(Metrics.class, metrics);
+
+        // add snowplow emitter to context, contributors can use this emitter to emit other event
+        emitterInit();
+        rootContext.put(Emitter.class,emitter);
+    }
+
+    private void emitterInit() {
+
+        // Make a new client with custom concurrency rules
+        PoolingHttpClientConnectionManager manager = new PoolingHttpClientConnectionManager();
+        manager.setDefaultMaxPerRoute(50);
+
+        // Make the client
+        CloseableHttpClient client = HttpClients.custom()
+                .setConnectionManager(manager)
+                .build();
+
+        // Build the adapter
+        HttpClientAdapter adapter = ApacheHttpClientAdapter.builder()
+                .url(TelemetryParams.TELEMETRY_SERVER_URL.toString())
+                .httpClient(client)
+                .build();
+
+        // Make a RequestCallback
+        RequestCallback callback = new RequestCallback() {
+
+            public void onSuccess(int successCount) {
+                logger.info("Success sent, successCount: " + successCount);
+            }
+
+            public void onFailure(int successCount, List<TrackerPayload> failedEvents) {
+                logger.warn("Failure, successCount: " + successCount + "\nfailedEvent:\n" + failedEvents.toString());
+            }
+        };
+
+        // initialise emitter
+        emitter = BatchEmitter.builder()
+                .httpClientAdapter(adapter) // Required
+                .threadCount(20) // Default is 50
+                .requestCallback(callback)
+                .bufferSize(1)
+                .build();
+    }
+
+    @Override
+    public void postInitialise(Context rootContext) {
+
+        metrics.initialise();
+        InjectionHelper.inject(this, rootContext);
```

Comment:
The InjectionHelper may be a bit too heavy for this.
You can just grab the config (only injected field) from the root context instead.